src/app/features/courses/components/search/search.component.ts

```diff
@@ -7,9 +7,13 @@ import { Component } from '@angular/core';
 })
 export class SearchComponent {
 
+  @Output() makeSearchQuery: EventEmitter<string> = new EventEmitter();
   searchQuery = '';
 
-  search() {
-    console.log(this.searchQuery);
+  search(event) {
```

Comment:
Pls add type here, and also i not sure, but `keypress` haven't type `click`. Pls check it

Reply:
The idea was to make search possible when clicking on search icon or when pressing 'Enter' button,  for better user experience. 
The search method was called when  search icon was clicked and when 'enter' key was pressed in the input field.
Made the following changes: 
- click on search icon calls search method;
- 'enter' key press calls handle keypress method, which checks, if 'Enter' was pressed.  
Event types added, where required.

Reply:
you can add few types here